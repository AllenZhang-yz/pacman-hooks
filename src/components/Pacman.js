import React from "react";

const Pacman = ({ direction }) => {
  return <div className={`pacman pacman--${direction.toLowerCase()}`}></div>;
};

export default Pacman;
