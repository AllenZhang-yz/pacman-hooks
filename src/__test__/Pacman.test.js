import React from "react";
import Pacman from "../components/Pacman";
import { shallow } from "enzyme";
import Enzyme from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import toJson from "enzyme-to-json";
Enzyme.configure({ adapter: new Adapter() });

const setup = ({ ...props }) => {
  const wrapper = shallow(<Pacman {...props} />);
  return {
    props,
    wrapper
  };
};

describe("Pacman", () => {
  const { wrapper } = setup({
    direction: "NORTH"
  });
  it("Pacman renders correctly", () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it("Pacman render direction correctly", () => {
    expect(wrapper.find("div").hasClass("pacman pacman--north")).toBe(true);
  });
});
