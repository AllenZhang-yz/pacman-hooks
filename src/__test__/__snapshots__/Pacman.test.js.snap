// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pacman Pacman renders correctly 1`] = `
<div
  className="pacman pacman--north"
/>
`;
